### PR TITLE
[RHACS] Updated Architecture guide

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://issues.redhat.com/browse/ROX-11254

Includes updated content and diagrams for the architecture topic.

Preview:https://openshift-docs-git-acs-architecture-gnelson.vercel.app/openshift-acs/master/architecture/acs-architecture.html

Merge into `rhacs-docs` and cherry-pick into:
- `rhacs-docs-3.74`
- `rhacs-docs-3.73`
- `rhacs-docs-3.72`
- `rhacs-docs-3.71`

---

NOTE:

- This PR is for RHACS docs.
- It doesn't require any special labels or milestones.

### CODE REVIEW

File: architecture/acs-architecture.adoc [hunk 1]
```diff
@@ -9,10 +9,17 @@ Discover {product-title} architecture and concepts.
...
+
+include::modules/acs-central-services-overview.adoc[leveloffset=+1]
+
+include::modules/acs-secured-cluster-services-overview.adoc[leveloffset=+1]
+
+include::modules/acs-architecture-external-components.adoc[leveloffset=+1]
+
+include::modules/acs-architecture-diffrences-ocp-kube.adoc[leveloffset=+1]
+
+include::modules/acs-architecture-interaction-between-services.adoc[leveloffset=+1]
```
Comment (mjpytlak) [inline architecture/acs-architecture.adoc:25]: While not directly related tot his PR, please update the meta data of this file to include `:content-type:`

**Review by mjpytlak — COMMENTED** [review]
@gaurav-nelson Couple of items that need addressing. In other cases, have a few suggestions. I was mainly confused by the addition of `acs-architecture-2.adoc`. I suspect that these are your working notes.

**Comment by gaurav-nelson** [comment]
> @gaurav-nelson Couple of items that need addressing. In other cases, have a few suggestions. I was mainly confused by the addition of `acs-architecture-2.adoc`. I suspect that these are your working notes.

🤦🏼 Yup. That was it, I forgot to delete it.

**Pull request merged**